doc/source/examples/dem/rectangular-hopper/rectangular-hopper.rst

```diff
@@ -0,0 +1,210 @@
+
+.. code-block:: text
+
+    subsection mesh
+        set type                                = gmsh
+        set file name                           = hopper90.msh
+        set initial refinement                  = 1
+        set expand particle-wall contact search = false
+        set check diamond cells                 = true
+    end
+
+.. figure:: images/packed_hopper.png
+    :width: 300
+    :alt: Mesh
+    :align: center
+
+    Rectangular hopper packed with particle before the discharge.
+
+Insertion info
+~~~~~~~~~~~~~~~~~~~
+
+An insertion box is defined inside and the top part of the hopper. The inserted number is choose to be a factor of the total number of particle. In this case, 14 insertion step is require to fill up the hopper with particles.
+
+.. code-block:: text
+
+    subsection insertion info
+        set insertion method                               = non_uniform
+        set inserted number of particles at each time step = 2910
+        set insertion frequency                            = 25000
+        set insertion box minimum x                        = -0.1030
+        set insertion box minimum y                        = 0.10644
+        set insertion box minimum z                        = .00224
+        set insertion box maximum x                        = 0.1030
+        set insertion box maximum y                        = 0.16020
+        set insertion box maximum z                        = 0.03136
+        set insertion distance threshold                   = 1.5
+        set insertion random number range                  = 0.1
+        set insertion random number seed                   = 20
+    end
+
+
+Lagrangian physical properties
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The total number of particles in this simulation is 40740 with a diameter of 2.24 mm.
+
+.. code-block:: text
+
+    subsection lagrangian physical properties
+        set gx                       = 0.0
+        set gy                       = -9.81
+        set gz                       = 0.0
+        set number of particle types = 1
+        subsection particle type 0
+            set size distribution type            = uniform
+            set diameter                          = 0.00224
+            set number                            = 40740
+            set density particles                 = 2500
+            set young modulus particles           = 1000000
+            set poisson ratio particles           = 0.3
+            set restitution coefficient particles = 0.94
+            set friction coefficient particles    = 0.2
+            set rolling friction particles        = 0.09
+        end
+        set young modulus wall           = 1000000
+        set poisson ratio wall           = 0.3
+        set friction coefficient wall    = 0.2
+        set restitution coefficient wall = 0.9
+        set rolling friction wall        = 0.09
+    end
+
+
+Model parameters
+~~~~~~~~~~~~~~~~~
+
+.. code-block:: text
```

Comment:
I would just add a quick comment on how you chose the load balance frequency and the neighborhood treshold. Otherwise if you are using default values, remove these lines :)

Reply:
Will refer to the silo example, almost the same, I changed the `dynamic contact search size coefficient` for no specific reasons